src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -165,7 +167,10 @@ async def trigger(self):
             )
 
             raw_results = self._raw_results_received.get(timeout=self.timeout_s)
-            LOGGER.info(f"Zocalo: found {len(raw_results['results'])} crystals.")
+            source_of_results = "CPU" if raw_results["ispyb_ids"]["gpu"] == None else "GPU"
```

Comment:
is `raw_results["ispyb_ids"]` a defaultdict? If not then won't this cause an `IndexError`? Possibly:
```suggestion
            source_of_results = "CPU" if not raw_results["ispyb_ids"].get("gpu") else "GPU"
```
(Although `"GPU" if raw_results["ispyb_ids"].get("gpu") else "CPU"` is more terse, the focus being on the presumed default of "CPU" might be justified)

Reply:
According to https://docs.python.org/3/library/stdtypes.html#dict.get , an ordinary dict returns None rather than raising an IndexError if the key doesn't exist.

Reply:
Also now out-of-date, but `== None` is normally not what you want, generally you want to `is None`. But the falseish comparison beats both.